src/OpenLoco/src/World/TownManager.cpp

```diff
@@ -29,12 +34,140 @@ namespace OpenLoco::TownManager
         return regs.eax;
     }
 
+    // 0x00497D70
+    static uint32_t copyTownNameToBuffer(const TownNamesObject* namesObj, uint32_t offset, uint16_t index, char* buffer)
+    {
+        auto* offsetPtr = reinterpret_cast<const std::byte*>(namesObj) + offset;
+        auto srcOffset = *reinterpret_cast<const int16_t*>(offsetPtr + index * 2);
+
+        auto* srcPtr = reinterpret_cast<const char*>(offsetPtr + srcOffset);
+        strcpy(buffer, srcPtr);
+
+        uint8_t tail = *reinterpret_cast<const uint8_t*>(srcPtr + strlen(srcPtr) + 1);
+        return tail;
+    }
+
+    // 0x00497A6A
+    static uint8_t townNameFromNamesObject(uint32_t& rand, const char* buffer)
+    {
+        auto* namesObj = ObjectManager::get<TownNamesObject>();
```

Comment:
We should add some comments about how the town name is composed of multiple categories concatenated together.
```
// {CAT1}{CAT2}{CAT3}{CAT4}{CAT5}{CAT6}
// Seperators are defined in the category string (so could be '-' or ' ' or nothing)
// Categories can be empty and can be skipped based on rand
// e.g. "Fort " + "Apple " + "Green" for "Fort Apple Green"
```